src/ucs/memory/numa.c

```diff
@@ -23,6 +35,9 @@ const char *ucs_numa_policy_names[] = {
     [UCS_NUMA_POLICY_LAST]      = NULL,
 };
 
+KHASH_MAP_INIT_INT(numa_distance, ucs_numa_distance_t);
+static khash_t(numa_distance) numa_distance_hash = KHASH_STATIC_INITIALIZER;
```

Comment:
WDYT of moving it to `ucs_topo_global_ctx` in `topo.c`? Then can utilized the same lock there.

Reply:
It's possible, but than you will get a dependency between numa implementation and topo, if you use numa_distance directly you will not have the cached results.
At the moment only topo uses the numa distance function but it the future it might change for some reason.